### PR TITLE
Close manifest cache DB before calling completion handlers

### PR DESCRIPTION
**Pull request by neonichu**
Previously, we were relying on `defer` to close the database, but that would mean we're closing it when exiting the scope which is typically *after* we are calling completion handlers. It seems as if we should instead call them before.

Related to rdar://101956252

### CODE REVIEW

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Review by abertelrud — APPROVED** [review]

**Comment by neonichu** [comment]
Hm, looks like this isn't working well: `Basics/SQLiteBackedCache.swift:66: Fatal error: db should be closed`

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Review by abertelrud — APPROVED** [review]

**Review by tomerd — APPROVED** [review]

File: Sources/PackageLoading/ManifestLoader.swift [hunk 1]
```diff
@@ -333,14 +333,17 @@ public final class ManifestLoader: ManifestLoaderProtocol {
             )
         }
 
-        var closeAfterRead = DelayableAction(target: cache) { cache in
```
Comment (tomerd) [inline Sources/PackageLoading/ManifestLoader.swift:336]: we may be able to deprecate DelayableAction with this, I think this was the only call site?

as a side note, all this code should be refactored to async/await to reduce complexity

**Review by tomerd — APPROVED** [review]

File: Sources/PackageLoading/ManifestLoader.swift (hunk 1 above)
Comment (neonichu) [reply Sources/PackageLoading/ManifestLoader.swift:336]: It's used in `evaluateManifest()` as well.

**Pull request merged**